cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/ApplyLogic.java

```diff
@@ -0,0 +1,57 @@
+package cloud.foundry.cli.logic;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import cloud.foundry.cli.crosscutting.logging.Log;
+import cloud.foundry.cli.crosscutting.mapping.beans.ApplicationBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.ConfigBean;
+import cloud.foundry.cli.logic.apply.ApplicationApplier;
+import cloud.foundry.cli.logic.apply.DiffWrapper;
+import cloud.foundry.cli.logic.diff.DiffNode;
+import cloud.foundry.cli.logic.diff.change.CfChange;
+import cloud.foundry.cli.operations.ApplicationsOperations;
+import org.cloudfoundry.operations.DefaultCloudFoundryOperations;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+
+public class ApplyLogic {
+
+    private DefaultCloudFoundryOperations cfOperations;
+
+    public ApplyLogic(@Nonnull DefaultCloudFoundryOperations cfOperations) {
+        checkNotNull(cfOperations);
+
+        this.cfOperations = cfOperations;
+    }
+
+    public void applyApplications(@Nonnull Map<String, ApplicationBean> desiredApplications) {
+        checkNotNull(desiredApplications);
+
+        ApplicationsOperations applicationsOperations = new ApplicationsOperations(cfOperations);
+        Map<String, ApplicationBean> liveApplications = applicationsOperations.getAll().block();
+
+        // that way only the applications of the live system are compared in the diff
+        ConfigBean desiredApplicationsConfig = new ConfigBean(desiredApplications);
+        ConfigBean liveApplicationsConfig = new ConfigBean(liveApplications);
+
+        // compare entire configs as the diff wrapper is only suited for diff trees of these
+        DiffLogic diffLogic = new DiffLogic();
+        DiffNode diffTreeRoot = diffLogic.createDiffTree(liveApplicationsConfig, desiredApplicationsConfig);
+        DiffWrapper wrappedDiff = new DiffWrapper(diffTreeRoot);
+
+        Map<String, List<CfChange>> allApplicationChanges = wrappedDiff.getApplicationChanges();
+
+        for (Entry<String, List<CfChange>> applicationChangesEntry : allApplicationChanges.entrySet()) {
+            String applicationName = applicationChangesEntry.getKey();
+            Log.debug("Start applying the changes to the app:", applicationName);
+            List<CfChange> applicationChanges = applicationChangesEntry.getValue();
+
+            ApplicationApplier.apply(new ApplicationsOperations(cfOperations), applicationName, applicationChanges);
```

Comment:
One problem that may arise is, that if there is an error while applying an application, the whole process terminates and it wouldn't apply any remaining apps. If this is intentional behavior, then never mind.

Reply:
The user story does not define the expected behavior in cases of errors. Thus we can leave it like that for now, but in the future we might indeed take this aspect into account.

Reply:
Yes I think this could be done in another US.